ledger/cow.go

```diff
@@ -185,7 +270,14 @@ func (cb *roundCowState) blockHdr(r basics.Round) (bookkeeping.BlockHeader, erro
 }
 
 func (cb *roundCowState) put(addr basics.Address, new basics.AccountData, newCreatable *basics.CreatableLocator, deletedCreatable *basics.CreatableLocator) {
-	cb.mods.Accts.Upsert(addr, new)
+	// convert AccountData to PersistentAccountData by using getPadCache
+	// that is must be filled in lookup* methods
+	if pad, ok := cb.getPadCache[addr]; ok {
+		pad.AccountData = new
+		cb.mods.Accts.Upsert(addr, pad)
+	} else {
+		panic(fmt.Sprintf("Address %s does not have entry in getPadCache", addr.String()))
```

Comment:
I know that this shouldn't happen.. but can we make it more transient without panicking ?
i.e. try to reload and recover, and log an error if we fail.
my concern is that this might be used as an attack vector agains the network.

Reply:
I implemented recovery option but it still can panic if no such address. I guess this is a valid behavior. Plus I had to change `lookupCreatableData` to accept multiple values at once to return since `new` might contain multiple entries that need to be loaded.